File: frontend/src/types/portfolio.ts
export interface OptimizationConfig {
   csv_type: string;
   tickers: string[];
   data_length: number;
   control_length: number;
   optimizer: string;
   window_size: number;
   hidden_size: number;
   samples_amount: number;
   risk_threshold: number;
   capital: number;
}

export interface Portfolio {
   risk: number;
   return: number;
   sharpe_ratio: number;
   tickers: string[];
   distribution: number[];
}

export interface ChartMetadata {
   x_axis_label: string;
   y_axis_label: string;
   color_bar_label: string;
   chart_title: string;
   x_range: [number, number];
   y_range: [number, number];
}

export interface OptimizationResponse {
   portfolios: Portfolio[];
   metadata: ChartMetadata;
}

File: backend/app/schemas/portfolio.py
from pydantic import BaseModel
from typing import List

class OptimizationRequest(BaseModel):
   tickers : List[str]
   data_length : int
   control_length : int
   window_size : int
   hidden_size : int
   samples_amount : int
   risk_threshold: int
   capital : int

